packages/lesswrong/components/tagging/CoreTagsChecklist.tsx

```diff
@@ -17,19 +18,21 @@ const styles = (theme: ThemeType): JssStyles => ({
     }
   },
   tag: {
-    minWidth: "25%",
-    display: "inline-block",
-    ...theme.typography.commentStyle,
-    marginRight: 16,
-    color: theme.palette.grey[600],
-    marginTop: 4
+    ...tagStyle(theme),
+    backgroundColor: "unset",
+    color: theme.palette.grey[500],
+    border: theme.palette.border.extraFaint,
+    '&:hover': {
+      border: theme.palette.border.grey300,
+      color: theme.palette.grey[800]
+    }
   }
-});
+}); 
 
-const CoreTagsChecklist = ({onSetTagsSelected, classes, post}: {
-  onSetTagsSelected: (selectedTags: Record<string,boolean>)=>void,
+const CoreTagsChecklist = ({onTagSelected, classes, existingTagIds=[] }: {
+  onTagSelected?: (props: {tagId: string, tagName: string})=>void,
   classes: ClassesType,
-  post: PostsList|SunshinePostsList
+  existingTagIds?: Array<string|undefined>
```

Comment:
Could we make this `existingTagIds: Array<string>`?

Reply:
I could fix it, but it requires either adding a filter on existingTagIds to remove any potential undefined ones, or, maybe remove the nullable option on tagRel.tagId. Nullable was added in this PR, I'm not sure why.

https://github.com/ForumMagnum/ForumMagnum/pull/3485

I don't think this is important enough to worry about for this PR, but seems good to flag it as a weird thing to think about later.

Reply:
:(

Sad, but I agree that it's not for this PR.